modules/@apostrophecms/ui/ui/apos/utils/index.js

```diff
@@ -1,9 +1,15 @@
 export const debounce = (func, timeout) => {
   let timer;
   return (...args) => {
-    clearTimeout(timer);
-    timer = setTimeout(() => {
-      func.apply(this, args);
-    }, timeout);
+    return new Promise((resolve, reject) => {
+      clearTimeout(timer);
+      timer = setTimeout(() => {
+        try {
+          resolve(func.apply(this, args));
```

Comment:
The only purpose of this change is to be able to await the debounced function and to get its result without changing the behavior of existing ones.
I agree for the try catch I didn't realized it had to await.
But the `resolve(func.apply(this, args))` works perfectly even if `func` is async.
I updated the code to make it work in case of errors too.